File: 6_linked_list_and_arrays/6_max_consecutive_ones.py
class Solution:
    def findMaxConsecutiveOnes(self, nums: List[int]) -> int:
        res = curr = 0
        for i,v in enumerate(nums):
            if v == 1:
                curr+=1
            else:
                res = max(res, curr)
                curr = 0
        return max(curr, res)

File: 07_linked_list_and_arrays/6_max_consecutive_ones.rs
use std::cmp::max;

impl Solution {
    pub fn find_max_consecutive_ones(nums: Vec<i32>) -> i32 {
        let (mut curr, mut res) = (0, 0);
        for &num in &nums {
            if num != 1 {
                res = max(res, curr);
                curr = 0;
            } else {
                curr += 1;
            }
        }
        max(res, curr)
    }
}
